scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat

```diff
@@ -3443,3 +3443,65 @@ object a:
       val x = 4
       4 == 3
    do 5
+<<< #3276
+class Foo() {
+  def ok: Boolean =
+    if (1 == 1) {
+      true
+    } else
+      false
+
+  def notOK: Boolean =
+    if (1 == 1) {
+      println("Some code")
+      true
+    } else
+      false
+}
+>>>
+class Foo() {
+  def ok: Boolean =
+    if (1 == 1) {
+      true
+    } else
+      false
+
+  def notOK: Boolean =
+    if (1 == 1) {
+      println("Some code")
+      true
+    } else
+      false
+}
+<<< #3276 2
+class Foo() {
+  def notOK: Boolean =
+    if (1 == 1)
+      {c; a + b}.foo
+    else
+      false
+}
+>>>
+class Foo() {
+  def notOK: Boolean =
+    if (1 == 1) { c; a + b }.foo
+    else
+      false
```

Comment:
```suggestion
  def notOK: Boolean =
    if (1 == 1) { c; a + b }.foo
    else false
```
why only fold thenp ?

Reply:
`else` part uses the existing newline. and the `then` part uses an old rule about `{`. i will copy the same tests into scala2, they are formatted exactly the same.

i will try to address that rule separately.

Reply:
@tgodzik addressed in #3282, ptal.